### PR TITLE
 [feature](nereids) support cast and extract date for TPC-H

### PR DESCRIPTION
**Pull request by yinzhijian**
# Proposed changes

Issue Number: close #xxx

## Problem Summary:

support cast and extract date for TPC-H, for example:
1. select cast(a as datetime) as d from test;
2. select extract(year from datetime_column) as y from test;

## Checklist(Required)

1. Does it affect the original behavior: (No)
4. Has unit tests been added: (Yes)
5. Has document been added or modified: (No Need)
6. Does it need to update dependencies: (No)
7. Are there any changes that cannot be rolled back: (No)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Cast.java [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+
+import java.util.List;
+
+/**
+ * cast function.
+ */
+public class Cast extends Expression implements BinaryExpression {
+
+    public Cast(Expression child, String type) {
+        super(ExpressionType.CAST, child, new StringLiteral(type));
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Cast.java:34]: maybe add a DataType attribute and let cast be a unary expression is better

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Year.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+public class Year extends BoundFunction implements UnaryExpression {
+
+    public Year(Expression child) {
+        super("year", child);
+    }
+
+    @Override
+    public DataType getDataType() {
+        return child().getDataType();
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Year.java:39]: year should return Int type, right?

**Comment by morrySnow** [comment]
please merge newest master

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Cast.java (hunk 1 above)
Comment (yinzhijian) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/Cast.java:34]: I have considered this, but this way WithChildren is not easy to implement

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Year.java (hunk 2 above)
Comment (yinzhijian) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/expressions/functions/Year.java:39]: yes

**Comment by yinzhijian** [comment]
> please merge newest master

done

**Review by morrySnow — APPROVED** [review]

**Review by 924060929 — APPROVED** [review]
LGTM

**Pull request merged**